candig/server/frontend.py

```diff
@@ -240,37 +240,46 @@ class UserAccessMap(object):
     Loads local authorization info from an access list file to the backend
     ACL tsv file header row contains list of project names(ID)
     """
-    def __init__(self):
+    def __init__(self, logger=None):
         self.user_access_map = {}
         self.file_path = app.config.get('ACCESS_LIST')
 
         if not self.file_path:
             raise exceptions.ConfigurationException("No user access list defined")
 
-        self.access_list = pd.read_csv(self.file_path, sep='\t')
+        try:
+            self.access_list = pd.read_csv(self.file_path, sep='\t', index_col=['issuer', 'username'])
+            # Detect duplicated (issuer, username) tuples
+            for issuer_user in self.access_list.index[self.access_list.index.duplicated()]:
+                raise ValueError(
+                    "Duplicate entries detected for {}. "
+                    "User access disabled until ACL resolved.".format(issuer_user))
+        except (IOError, ValueError) as err:
+            self.access_list = None
+            if logger:
+                logger.error(err)
+
         self.list_updated = os.path.getmtime(self.file_path)
 
-    def initializeUserAccess(self, logger=None):
-        project_list = list(self.access_list)[1:]
-        for idx, row in self.access_list.iterrows():
-            username = row['users']
-            if username not in self.user_access_map:
-                self.user_access_map[username] = {}
-                for project in project_list:
-                    try:
-                        level = int(row[project])
-                        if 0 <= level <= 4:
-                            self.user_access_map[username][project] = level
-                    except ValueError:
-                        continue
-            else:
-                if logger:
-                    logger.warn(
-                        "Duplicate entries detected for {}. Resolve in ACL.".format(username)
-                    )
+    def initializeUserAccess(self):
+        # Convert user access table into a dictionary
+        if self.access_list is not None:
+            self.user_access_map = self.access_list.to_dict(orient='index')
+
+        # Remove non set values
+        self.user_access_map = {
+            user: {project: level
+                   for project, level in value.iteritems() if 0 <= level <= 4
+                   }
+            for user, value in self.user_access_map.iteritems()
```

Comment:
Is this a problem when the `self.user_access_map` is of `NoneType`? `None.iteritems()` could potentially be problematic?

Reply:
I don't think it's possible for it to be None. Default init is {}

Reply:
Ah right, I didn't check the `init` function.